VCCrypto/VCCrypto/CryptoOperations.swift

```diff
@@ -3,78 +3,56 @@
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
-public protocol CryptoOperating {
-    func generateKey() throws -> VCCryptoSecret
-    func retrieveKeyFromStorage(withId id: UUID) -> VCCryptoSecret
-    func save(key: Data, withId id: UUID) throws
-    func deleteKey(withId id: UUID) throws
-    func getKey(withId id: UUID) throws -> Data
+enum CryptoOperationsError: Error {
+    case invalidPublicKey
 }
 
+/// Operations that are involved in verification cryptographic operations..
 public struct CryptoOperations: CryptoOperating {
-
-    private let secretStore: SecretStoring
     
-    private let sdkConfiguration: VCSDKConfigurable
+    public init() {}
     
-    public init(sdkConfiguration: VCSDKConfigurable) {
-        self.init(secretStore: KeychainSecretStore(), sdkConfiguration: sdkConfiguration)
+    /// Only supports Secp256k1 signing.
+    public func sign(messageHash: Data, usingSecret secret: VCCryptoSecret) throws -> Data  {
+        let algorithm = try Secp256k1(secret: secret)
+        return try algorithm.sign(messageHash: messageHash)
     }
     
-    public init(secretStore: SecretStoring, sdkConfiguration: VCSDKConfigurable) {
-        self.secretStore = secretStore
-        self.sdkConfiguration = sdkConfiguration
+    public func hash(message: Data, algorithm: SupportedHashAlgorithm) -> Data {
+        switch algorithm {
+        case .SHA256:
+            return Sha256().hash(data: message)
+        case .SHA512:
+            return Sha512().hash(data: message)
+        }
     }
     
-    public func generateKey() throws -> VCCryptoSecret {
-        let accessGroup = sdkConfiguration.accessGroupIdentifier
-        let key = try Random32BytesSecret(withStore: secretStore, inAccessGroup: accessGroup)
-        return key
+    /// Only support Secp256k1 public key retrieval.
+    public func getPublicKey(fromSecret secret: VCCryptoSecret) throws -> PublicKey {
+        return try Secp256k1(secret: secret).getPublicKey()
     }
     
-    public func retrieveKeyFromStorage(withId id: UUID) -> VCCryptoSecret {
-        let accessGroup = sdkConfiguration.accessGroupIdentifier
-        return Random32BytesSecret(withStore: secretStore, andId: id, inAccessGroup: accessGroup)
+    public func verify(signature: Data,
+                       forMessage message: Data,
+                       usingPublicKey publicKey: PublicKey) throws -> Bool {
+        
+        let algorithm = try getAlgorithm(publicKey: publicKey)
+        
+        /// If algorithm is secp256k1, hash message value.
+        var msg = message
```

Comment:
Should this be done inside the secp256k class?

Reply:
per discussion created a ES256k class that does hashing and secp256k1 signing